enhancements/console/multi-cluster-console.md

```diff
@@ -0,0 +1,366 @@
+`openshift-console-user-settings` namespace. In a multi-cluster environment, we
+will continue to use these config maps and always store the user settings on
+the hub. Most user settings are not specific a particular cluster.
+
+For settings that are cluster-specific, they will need to be set by cluster
+name under a `cluster-settings` stanza in the settings JSON. For example,
+
+```json
+{
+  "cluster-settings": {
+    "local-cluster": {
+      "console.lastNamespace": "my-namespace"
+    }
+  }
+}
+```
+
+### User Stories
+
+See [Hybrid Console personas, use cases, and principles](https://docs.google.com/document/d/1-UodIqKYIsOaIYVNYNqpGWu7PnRvtml_nn1EpvnsSVo/edit#).
+
+### Implementation Details/Notes/Constraints [optional]
+
+* This proposal assumes the console backend running on the hub cluster will be
+  able to make network requests to public endpoints like the API server on each
+  spoke cluster.
+* Dynamic plugins will only be loaded from the hub cluster. If a dynamic plugin
+  is added by an operator on a spoke, it will not be loaded by the
+  multi-cluster console on the hub. We'll need additional exploration to see if
+  loading dynamic plugins from a spoke can be supported.
+* The console service account will not have special permissions on the spoke
+  clusters. This means a few features like cloud shell that require special
+  access will need to be disabled for multi-cluster.
+
+### Risks and Mitigations
+
+This is a significant architectural change to OpenShift console, and the
+enhancement proposal makes assumptions about whether the console backend on a
+hub cluster can reach public endpoints on various clusters in the fleet. Some
+capabilities like cloud shell require special privileges from the console
+service account, which won't work in a multi-cluster environment.
+
+We are mitigating these risks by initially offering this as a tech preview as we
+collect feedback and work through technical limitations.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Will the console running on the hub always be able to talk to public API
+   endpoints on the spoke cluster?
+1. What is our long-term approach to multi-cluster metrics? Will workload
+   metrics from all clusters the fleet be available from Thanos on the hub
+   cluster?
+1. Do we need to be able to load dynamic plugins from spoke clusters?
+
+### Test Plan
+
+For the tech preview, we will need to rely on unit tests for multi-cluster
+capability. openshift/console CI currently is not set up to provision a real
+multi-cluster environment.
+
+End-to-end tests that cover multi-cluster scenarios will be a GA graduation
+requirement.
+
+### Graduation Criteria
+
+We will initially deliver this as a tech preview.
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default when ACM is installed
+- Ability to connect to non-OpenShift clusters in the fleet
```

Comment:
isnt this in the **Non-Goals** section ?

Reply:
Yeah, ultimately we want to support this, just not for the tech preview. I'll take it out of non-goals.